### PR TITLE
TW-1926: Permission is always asked when paste in iOS

### PR DESCRIPTION
**Pull request by sherlockvn**
## Ticket need to merge:
https://github.com/linagora/flutter_contacts/pull/11

## Ticket
**Related issue**
#1926 
## Solution
Reference from here: https://github.com/flutter/flutter/issues/103163
Flutter have fixed the problem, and it work super well.
We need to upgrade flutter to latest to have this new bug fix: flutter: 3.24.0
## Resolved
**Attach screenshots or videos demonstrating the changes**
- Web:

https://github.com/user-attachments/assets/20477340-2d2f-4dbb-b73e-b046ad6265ec

- Android:
- IOS:

https://github.com/user-attachments/assets/4e70c6e9-dd56-4ab2-b5e6-c7c112645488

### Image save function: 

https://github.com/user-attachments/assets/55fe77ab-f109-44c1-8b68-9bd0b692316a

https://github.com/user-attachments/assets/a46d9a7f-e870-4662-a23d-727400c17295

### Sharing function:

https://github.com/user-attachments/assets/f2490a10-cbab-42ed-ac66-756de47e619f

https://github.com/user-attachments/assets/85a22b82-0938-4fcb-9670-fb213be755d0

### Notification

https://github.com/user-attachments/assets/2317418e-6025-4806-a54b-513a0a68989e

### CODE REVIEW

**Comment by Te-Z** [comment]
Hi @sherlockvn I found this repository: https://gitlab.com/linagora . IMO you should ask Dat to add the gitlab repositories to it WDYT ?

**Comment by nqhhdev** [comment]
> Hi @sherlockvn I found this repository: https://gitlab.com/linagora . IMO you should ask Dat to add the gitlab repositories to it WDYT ?

IMO: Create repositories in github

File: lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body_view.dart [hunk 1]
```diff
@@ -59,6 +59,7 @@ class AppAdaptiveScaffoldBodyView extends StatelessWidget {
         builder: (context, activeNavigationBar, __) {
           return PopScope(
             canPop: activeNavigationBar == AdaptiveDestinationEnum.rooms,
+            // ignore: deprecated_member_use
```
Comment (nqhhdev) [inline lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body_view.dart:62]: Why need to ignore in here?

File: ios/Runner/AppDelegate.swift [hunk 2]
```diff
@@ -3,7 +3,7 @@ import Flutter
 
 let apnTokenKey = "apnToken"
 
-@UIApplicationMain
+@main
```
Comment (nqhhdev) [inline ios/Runner/AppDelegate.swift:6]: Pls write ADR or more information for this change

File: pubspec.yaml [hunk 3]
```diff
@@ -16,8 +16,8 @@ dependencies:
 
   receive_sharing_intent:
     git:
-      url: git@github.com:krabbenprgr/receive_sharing_intent.git
-      ref: receive-.txt-v2
+      url: https://github.com/sherlockvn/receive_sharing_intent.git
+      ref: master
```
Comment (nqhhdev) [inline pubspec.yaml:20]: Pls use lib in linagora repo

https://github.com/linagora/receive_sharing_intent

Comment (sherlockvn) [reply pubspec.yaml:20]: done

File: lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body_view.dart (hunk 1 above)
Comment (sherlockvn) [reply lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body_view.dart:62]: https://docs.flutter.dev/release/breaking-changes/navigator-and-page-api

Comment (sherlockvn) [reply lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body_view.dart:62]: This change need to verified carefully, so i think we merged it later for testing

**Comment by sherlockvn** [comment]
i tried to run CI for the branch, and only debug version of window failed. here is link to the issue: https://github.com/firebase/flutterfire/issues/12051
image

File: docs/adr/0024-upgrade-flutter-3.24.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+## Context
+
+To enable features like text pasting without requiring user permission on iOS, we need to upgrade to Flutter 3.24.0.
+
+## Decision
+
+During the upgrade to Flutter 3.24.0, some dependencies must also be updated to ensure successful release builds. Specifically, the `targetSdkVersion` or `compileSdkVersion` needs to be updated to at least 31. Below is the list of packages that require a minimum `compileSdkVersion` of 31 or higher for successful compilation:
+
+- **`callkeep`**: Upgrade `compileSdkVersion` from 28 to 31. [See PR #190](https://github.com/flutter-webrtc/callkeep/pull/190)
```
Comment (nqhhdev) [inline docs/adr/0024-upgrade-flutter-3.24.md:17]: Currently, we're using android 14 (compileSdkVersion is 34) I think should update also

File: docs/adr/0024-upgrade-flutter-3.24.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+- **`receive_sharing_intent`**: Switch to the version maintained by Linagora, which has `compileSdkVersion` set to 33. [See Linagora Repository](https://github.com/linagora/receive_sharing_intent)
+- **`flutter_contacts`**: Upgrade `compileSdkVersion` from 31 to 33. [See PR #11](https://github.com/linagora/flutter_contacts/pull/11)
```
Comment (nqhhdev) [inline docs/adr/0024-upgrade-flutter-3.24.md:19]: Currently, we're using android 14 (compileSdkVersion is 34) I think should update also

Comment (sherlockvn) [reply docs/adr/0024-upgrade-flutter-3.24.md:19]: We don't have to, when the author update it, we will follow

image

Comment (nqhhdev) [reply docs/adr/0024-upgrade-flutter-3.24.md:19]: Pls deep thoughts!

Source: https://developer.android.com/google/play/requirements/target-sdk

File: docs/adr/0024-upgrade-flutter-3.24.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+**Date:** 2024-08-26
+
+## Status
+
+**Accepted**
+
+## Context
+
+To enable features like text pasting without requiring user permission on iOS, we need to upgrade to Flutter 3.24.0.
```
Comment (hoangdat) [inline docs/adr/0024-upgrade-flutter-3.24.md:11]: can you ref the link for this or the change log of Flutter indicate this

File: docs/adr/0024-upgrade-flutter-3.24.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+## Decision
+
+During the upgrade to Flutter 3.24.0, some dependencies must also be updated to ensure successful release builds. Specifically, the `targetSdkVersion` or `compileSdkVersion` needs to be updated to at least 31. Below is the list of packages that require a minimum `compileSdkVersion` of 31 or higher for successful compilation:
```
Comment (hoangdat) [inline docs/adr/0024-upgrade-flutter-3.24.md:15]: IMO, we dont need to upgrade `targetSdkVersion` here

**Comment by hoangdat** [comment]
- [ ] please have demo in Android
- [ ] demo receive sharing
- [ ] regression test app (because upgrade version)

File: docs/adr/0024-upgrade-flutter-3.24.md (hunk 5 above)
Comment (hoangdat) [reply docs/adr/0024-upgrade-flutter-3.24.md:19]: IMO, please update to 34

File: pubspec.yaml [hunk 8]
```diff
@@ -163,7 +179,11 @@ dependencies:
   async: ^2.11.0
   cached_network_image: ^3.2.3
   flutter_image_compress: ^2.0.4
-  image_gallery_saver: ^2.0.3
+  # TODO: remove when https://github.com/hui-z/image_gallery_saver/pull/310 is merged
+  image_gallery_saver:
```
Comment (hoangdat) [inline pubspec.yaml:183]: - [ ] demo functions related to image_saver

File: pubspec.yaml [hunk 9]
```diff
@@ -77,23 +81,31 @@ dependencies:
...
+  flutter_app_badger:
+    git:
+      url: https://github.com/bitsydarel/flutter_app_badger.git
+      ref: master
   flutter_app_lock: ^3.0.0
   flutter_blurhash: ^0.8.2
   flutter_cache_manager: ^3.3.0
   flutter_foreground_task: ^3.10.0
-  flutter_local_notifications: ^17.1.2
+  flutter_local_notifications: 17.2.2
```
Comment (hoangdat) [inline pubspec.yaml:101]: - [ ] demo related to notification in Android

File: pubspec.yaml [hunk 10]
```diff
@@ -77,23 +81,31 @@ dependencies:
   emoji_picker_flutter: ^1.5.1
   emoji_proposal: ^0.0.1
   emojis: ^0.9.9
-  fcm_shared_isolate: ^0.1.0
+  # TODO: remove this after this PR is merged https://gitlab.com/famedly/company/frontend/libraries/native_imaging/-/merge_requests/22
+  fcm_shared_isolate:
+    git:
+      url: https://github.com/famedly/fcm_shared_isolate.git
+      ref: main
```
Comment (hoangdat) [inline pubspec.yaml:88]: - [ ] demo notification in iOS

File: pubspec.yaml [hunk 11]
```diff
@@ -16,8 +16,8 @@ dependencies:
 
   receive_sharing_intent:
     git:
-      url: git@github.com:krabbenprgr/receive_sharing_intent.git
-      ref: receive-.txt-v2
+      url: https://github.com/linagora/receive_sharing_intent.git
+      ref: master
```
Comment (hoangdat) [inline pubspec.yaml:20]: - [ ] demo function in sharing

Comment (sherlockvn) [reply pubspec.yaml:20]: iOS
https://github.com/user-attachments/assets/21dd2231-0baa-48f4-88b1-49fa411d16b1

Comment (sherlockvn) [reply pubspec.yaml:20]: iOS + link + text: 

https://github.com/user-attachments/assets/f2490a10-cbab-42ed-ac66-756de47e619f

File: pubspec.yaml (hunk 8 above)
Comment (sherlockvn) [reply pubspec.yaml:183]: iOS:

https://github.com/user-attachments/assets/55fe77ab-f109-44c1-8b68-9bd0b692316a

**Comment by sherlockvn** [comment]
- [ ] Recheck Dialog download file

File: pubspec.yaml [hunk 12]
```diff
@@ -77,23 +81,31 @@ dependencies:
...
+      ref: main
   file_picker: ^8.0.5
   flutter:
     sdk: flutter
-  flutter_app_badger: ^1.5.0
+  # TODO: remove this after this PR is merged https://github.com/g123k/flutter_app_badger/pull/92
+  flutter_app_badger:
+    git:
+      url: https://github.com/bitsydarel/flutter_app_badger.git
+      ref: master
```
[the remark this answers is not in the document]
Comment (nqhhdev) [reply pubspec.yaml:96]: Not work for my side

[the remark this answers is not in the document]
Comment (nqhhdev) [reply pubspec.yaml:96]: Currently app badger on iOS is also not working properly.  Improve it on this ticket: https://github.com/linagora/twake-on-matrix/issues/1848

[the remark this answers is not in the document]
Comment (sherlockvn) [reply pubspec.yaml:96]: image
badger still works like old version in iOS

File: pubspec.yaml (hunk 9 above)
Comment (sherlockvn) [reply pubspec.yaml:101]: @nqhhdev hi, can you add me the demo of noti in android, please

Comment (sherlockvn) [reply pubspec.yaml:101]: thanks you

File: pubspec.yaml (hunk 11 above)
Comment (sherlockvn) [reply pubspec.yaml:20]: @nqhhdev can you help me add the demo of android in this case also, please ?

Comment (nqhhdev) [reply pubspec.yaml:20]: DONE !

File: pubspec.yaml (hunk 9 above)
Comment (nqhhdev) [reply pubspec.yaml:101]: Done !

File: pubspec.yaml (hunk 10 above)
Comment (nqhhdev) [reply pubspec.yaml:88]: Any update @sherlockvn ?

Comment (sherlockvn) [reply pubspec.yaml:88]: https://github.com/user-attachments/assets/735fa7ca-f25d-4d53-b14f-53a97c82e61b

File: pubspec.yaml (hunk 8 above)
Comment (sherlockvn) [reply pubspec.yaml:183]: https://github.com/user-attachments/assets/d24128bf-37c9-4b58-9fd8-f2be2efde1bb

**Review by nqhhdev — APPROVED** [review]
LGTM

File: pubspec.yaml [hunk 13]
```diff
@@ -38,7 +38,7 @@ dependencies:
   contacts_service:
     git:
       url: git@github.com:linagora/flutter_contacts.git
-      ref: master
+      ref: upgrade-targetAndCompileSdkVersion
```
Comment (hoangdat) [inline pubspec.yaml:41]: Merged. please update it

File: pubspec.yaml (hunk 9 above)
Comment (hoangdat) [reply pubspec.yaml:101]: how can i find the demo?

**Review by hoangdat — APPROVED** [review]

**Pull request merged**